src/gui/accountstate.cpp

```diff
@@ -443,4 +400,62 @@ std::unique_ptr<QSettings> AccountState::settings()
     return s;
 }
 
+namespace {
+
+    class UpdateUrlDialog : public QMessageBox
+    {
+    public:
+        UpdateUrlDialog(AccountPtr account, const QUrl &newUrl, QWidget *parent)
+            : QMessageBox(QMessageBox::Warning,
+                tr("Url update requested for %1").arg(account->displayName()),
+                tr("The url for %1 changed from %2 to %3, do you want to accept the changed url?").arg(account->displayName(), account->url().toString(), newUrl.toString()),
+                QMessageBox::NoButton, parent)
+            , _account(account)
+            , _newUrl(newUrl)
+        {
+            setAttribute(Qt::WA_DeleteOnClose);
+            addButton(tr("Change url permanently to %1").arg(newUrl.toString()), QMessageBox::AcceptRole);
+            addButton(tr("Reject"), QMessageBox::RejectRole);
+            connect(this, &QMessageBox::accepted, this, &UpdateUrlDialog::acceptNewUrl);
+        }
+
+    public slots:
+        void acceptNewUrl()
+        {
+            _account->setUrl(_newUrl);
+            Q_EMIT _account->wantsAccountSaved(_account.data());
+        }
+
+    private:
+        AccountPtr _account;
+        const QUrl _newUrl;
+    };
+
+} // anonymous namespace
+
+void AccountState::updateUrlDialog(AccountPtr &account, const QUrl &newUrl)
+{
+    auto updateUrlDialog = new UpdateUrlDialog(account, newUrl, ocApp()->gui()->settingsDialog());
```

Comment:
What about https://github.com/owncloud/client/pull/9184/commits/b7d82f4539df83b59bc3d3ebbaeea332139bb25f#diff-feb647b6021a1b6be5318e112187bd9e0ed390397dfe2f39d1b1e6c52c642473L37

I think it was one of the reasons why I moved the dialog to the account state.